mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -201,7 +202,7 @@ def Rock_ReduceOp :
 def Rock_ZeroInitKernelOp :
     Rock_Op<"zero_init_kernel", []>,
     Arguments<(ins AnyTensorOrMemRef:$buffer,
-                  StrAttr:$arch,
```

Comment:
Don't remove `arch` here - it may still be useful

Reply:
ok, so two points here : 
* We dont use it as of now -- nothing breaks if I remove it.
* Should we be really passing the string target-triple ? especially given that we have the bag of arch subfeatures (i.e. GemmFeatures) that could be passed instead. I really think we should not be putting in code in lowering : 
```
if arch == "gfxXXX": 
  do something
else if arch == "gfxYYY":
  do something else
```

Reply:
We will be putting that in when making decisions about which xdlops are available

Reply:
ok, please help me understand this : 

right now If I do a grep for "arch" or "Arch" in the Rock/ : 

```
manupa@manupa-System-Product-Name:~/work/rocMLIR/mlir/lib/Dialect/Rock$ grep "arch" . -r
./Generator/AmdArchDb.cpp:AmdArchInfo mlir::rock::lookupArchInfo(StringRef arch) {
./Generator/AmdArchDb.cpp:  std::tie(firstPart, remainingParts) = arch.split(':');
./Generator/AmdArchDb.cpp:  llvm::errs() << "Warning: unknown architecture, falling back to defaults: "
./Generator/AmdArchDb.cpp:               << arch << "\n";
./Generator/Conv2dGenerator.cpp:    const std::string &arch, const std::string &chip, const std::string &triple,
./Generator/Conv2dGenerator.cpp:    : config{arch,
./Generator/Conv2dGenerator.cpp:  std::string arch;
./Generator/Conv2dGenerator.cpp:  strToStr("arch", arch);
./Generator/Conv2dGenerator.cpp:  if (failed(splitter.parse(arch))) {
./Generator/Conv2dGenerator.cpp:  // Canonicalize architecture name
./Generator/Conv2dGenerator.cpp:  arch = canonicalArch.str();
./Generator/Conv2dGenerator.cpp:  config.arch = arch;
./Generator/Conv2dGenerator.cpp:  AmdArchInfo archInfo = lookupArchInfo(splitter.getChip());
./Generator/Conv2dGenerator.cpp:  config.features = archInfo.defaultFeatures;
./Generator/Conv2dGenerator.cpp:  StringAttr archStrAttr = builder.getStringAttr(config.arch);
./Generator/Conv2dGenerator.cpp:  NamedAttribute archAttr = builder.getNamedAttr("xmodel.arch", archStrAttr);
./Generator/Conv2dGenerator.cpp:      archAttr};
./Generator/Conv2dGenerator.cpp:    module->setAttr(archAttr.getName(), archAttr.getValue());
./Generator/Conv2dGenerator.cpp:      builder.getNamedAttr("arch", archStrAttr),
./Tuning/ConvContext.cpp:  auto archVal = op->getAttrOfType<StringAttr>("arch").getValue();
./Tuning/ConvContext.cpp:  return {archVal,     numCu,      opType, dimIndexAndSize, strideVal,
./Tuning/RockTuningImpl.cpp:// Brute-force search in incremental order
./Tuning/RockTuningImpl.cpp:  auto search = perfTable->tuningMap.find(problem);
./Tuning/RockTuningImpl.cpp:  if (search != perfTable->tuningMap.end()) {
./Tuning/RockTuningImpl.cpp:  auto search = perfTable->tuningMap.find(problem);
./Tuning/RockTuningImpl.cpp:  if (search != perfTable->tuningMap.end()) {
manupa@manupa-System-Product-Name:~/work/rocMLIR/mlir/lib/Dialect/Rock$ grep "Arch" . -r
./Generator/CMakeLists.txt:  AmdArchDb.cpp
./Generator/AmdArchDb.cpp://===- AmdArchDb.cpp - Dtabase of AMD GPU features ------------------===//
./Generator/AmdArchDb.cpp:#include "mlir/Dialect/Rock/Generator/AmdArchDb.h"
./Generator/AmdArchDb.cpp:static constexpr AmdArchInfo gcnInfo(GemmFeatures::none, /*waveSize=*/64),
./Generator/AmdArchDb.cpp:AmdArchInfo mlir::rock::lookupArchInfo(StringRef arch) {
./Generator/AmdArchDb.cpp:    return llvm::StringSwitch<AmdArchInfo>(minor)
./Generator/AmdArchDb.cpp:        .Case("06", AmdArchInfo(GemmFeatures::dot, /*waveSize=*/64))
./Generator/AmdArchDb.cpp:    return llvm::StringSwitch<AmdArchInfo>(minor)
./Generator/Conv2dGenerator.cpp:#include "mlir/Dialect/Rock/Generator/AmdArchDb.h"
./Generator/Conv2dGenerator.cpp:  SmallString<64> canonicalArch;
./Generator/Conv2dGenerator.cpp:  splitter.getFullName(canonicalArch);
./Generator/Conv2dGenerator.cpp:  arch = canonicalArch.str();
./Generator/Conv2dGenerator.cpp:  AmdArchInfo archInfo = lookupArchInfo(splitter.getChip());
./Tuning/RockTuningImpl.cpp:  problemOS << gemmIF.getArch() << tab;
./Transforms/ConvToGemm.cpp:      /*cTransposed=*/nullptr, op.getArchAttr(), op.getNumCuAttr(),
./Transforms/ConvToGemm.cpp:      /*cTransposed=*/nullptr, op.getArchAttr(), op.getNumCuAttr(),
./Transforms/ConvToGemm.cpp:                     /*cTransposed=*/nullptr, op.getArchAttr(),
```

"arch" only seems to be present in generators and rock::GemmOp. So generators parse the arch and generate features and use them throughout the lowering (rightfully so). rock::GemmOp only needed it because GridwiseGemm needed it and so on -- but none of the lower level ops seems never really used it. 

Are you talking about an upcoming feature ?

Reply:
Yeah, I'm talking about the upcoming need to, for example, not select fp8 when we're not on MI-300, or, more usefully, the ability to select the MI-200+ xdlops when we're on something newer than MI-100

Reply:
Hmmm, why cannot we resolve the target-triple to features beforehand for those (a change in AmdArchDB)? -- rather than having to pass around arch string to decide in the lowering.

(Roughly being analogous to target features of AMDGPU backend of LLVM)

Reply:
My concern is AMDGPU arch versions will add and drop features (as we are seeing with fmax for e.g.) and I personally feely best way to control the impact on the our lowering is to resolve the arch string to features and only use those in the lowering. WDYT ?

Reply:
It's not an unreasonable decision, though I do think things like the xdlops database might want to be keyed by arch.

(You'll note, for example, that the `amdgpu` dialect has `chipset` and does a lot of `chipset.minorVersion >= 0x0a` type stuff)

Though ... yeah, having stuff like "bf16_mfma" and "fp8" gated by features is also sensible, and we could key the xdlops selector by feature instead.

... I'll approve dropping `arch` as an attribute, yeah.